Rubberduck.Parsing/Binding/Bindings/DictionaryAccessDefaultBinding.cs

```diff
@@ -0,0 +1,218 @@
+        {
+            if (Tokens.Variant.Equals(asTypeName, StringComparison.InvariantCultureIgnoreCase) 
+                    || Tokens.Object.Equals(asTypeName, StringComparison.InvariantCultureIgnoreCase))
+            {            
+                /*
+                    The declared type of <l-expression> is Object or Variant. 
+                    In this case, the dictionary access expression is classified as an unbound member with 
+                    a declared type of Variant, referencing <l-expression> with no member name. 
+                */
+                ResolveArgumentList(null, argumentList);
+                return new DictionaryAccessExpression(null, ExpressionClassification.Unbound, expression, lExpression, argumentList);
+            }
+
+            /*
+                The declared type of <l-expression> is a specific class, which has a public default Property 
+                Get, Property Let, function or subroutine.
+            */
+            var defaultMember = (asTypeDeclaration as ClassModuleDeclaration)?.DefaultMember;
+            if (defaultMember == null
+                || !IsPropertyGetLetFunctionProcedure(defaultMember)
+                || !IsPublic(defaultMember))
+            {
+                ResolveArgumentList(null, argumentList);
+                return CreateFailedExpression(lExpression, argumentList);
+            }
+
+            var defaultMemberClassification = DefaultMemberClassification(defaultMember);
+
+            var parameters = ((IParameterizedDeclaration) defaultMember).Parameters.ToList();
+
+            if (IsCompatibleWithOneStringArgument(parameters))
+            {
+                /*
+                    This default member’s parameter list is compatible with <argument-list>. In this case, the 
+                    dictionary access expression references this default member and takes on its classification and 
+                    declared type.  
+                */
+                ResolveArgumentList(defaultMember, argumentList);
+                return new DictionaryAccessExpression(defaultMember, defaultMemberClassification, expression, lExpression, argumentList);
+            }
+
+            if (parameters.Count(param => !param.IsOptional) == 0 
+                && DEFAULT_MEMBER_RECURSION_LIMIT > recursionDepth)
+            {
+                /*
+                    This default member cannot accept any parameters. In this case, the static analysis restarts 
+                    recursively, as if this default member was specified instead for <l-expression> with the 
+                    same <argument-list>.
+                */
+
+                return ResolveRecursiveDefaultMember(defaultMember, defaultMemberClassification, argumentList, expression, recursionDepth);
+            }
```

Comment:
Great job with the comments there 👍

Reply:
They are basically copied from `IndexDefaultBinding`, where autoboosh added them. I just modified them to refer to a dictionary access expression instead of an index expression.